discovery_engine_core/web-api/src/personalization/routes.rs

```diff
@@ -85,24 +85,30 @@ async fn update_interactions(
         .map(|document| &document.document_id)
         .collect_vec();
     let documents = state.elastic.get_documents_by_ids(&ids).await?;
-    let embeddings = documents
-        .into_iter()
-        .map(|document| (document.id, document.embedding))
+    let documents = documents
+        .iter()
+        .map(|document| (&document.id, document))
         .collect::<HashMap<_, _>>();
 
     for document in interactions.documents {
         match document.interaction_type {
             UserInteractionType::Positive => {
-                //TODO for some reason this was returning a BAD_REQUEST error????
-                state
-                    .db
-                    .update_positive_cois(&document.document_id, &user_id, |positive_cois| {
-                        state.coi.log_positive_user_reaction(
-                            positive_cois,
-                            &embeddings[&document.document_id],
-                        )
-                    })
-                    .await?;
+                if let Some(document) = documents.get(&document.document_id) {
```

Comment:
this fixes a panic where the set of fetched documents/embeddings is smaller than the set of interacted documents.